scipy/sparse/linalg/isolve/minres.py

```diff
@@ -85,8 +85,6 @@ def minres(A, b, x0=None, shift=0.0, tol=1e-5, maxiter=None, xtype=None,
 
     eps = finfo(xtype).eps
 
-    x = zeros(n, dtype=xtype)
```

Comment:
This is probably not sufficient, you likely also need to other changes to the algorithm.
In principle `b = b - matvec(x)` does almost the right thing, but it messes up the tolerance computations, so the change to the algorithm is likely more complex.